src/common/wallet-state.ts

```diff
@@ -99,7 +85,25 @@ interface SendTransactionParams {
   nonce: number
 }
 
-export interface LoadedState {
+// States
+
+interface CommonState {
+  tncAccepted: boolean
+  setTncAccepted: (value: boolean) => void
```

Comment:
Isn't it only relevant in NoWalletState?

Reply:
Well...right now it is displayed only as a part of the wallet setup, but it doesn't mean that it does not make sense in other states. I can imagine that in the future when TnC are updated, we will want to display them to all users regardless of the state.

Reply:
If the T&C changed, it doesn't matter if they accepted the previous one.

Reply:
Hm, we still need a way to persist the acceptance of the new T&Cs in that case, so let's keep it this way.
We'll need a migration if it ever changes.

Reply:
Yeah, that was exactly my thinking :)